File: SQLite/example/ch01/examples/ruby/parameters.rb
#!/usr/bin/env ruby

require 'sqlite3'

db = SQLite3::Database.new("seinfeld.db")
db.results_as_hash = true

# Named paramters

stmt = db.prepare('SELECT * FROM foods where name like :name')
stmt.bind_param(':name', '%Peach%')

stmt.execute() do |result|
  result.each do |row|
    puts row['name']
  end
end

# Positional paramters

stmt = db.prepare('SELECT * FROM foods where name like ?')
stmt.bind_params('%Bobka%')

stmt.execute() do |result|
  result.each do |row|
    puts row['name']
  end
end

# Free read lock
stmt.close()

File: SQLite/example/ch01/examples/perl/parameters.pl
#!/usr/bin/env perl

use DBI;

print "Drivers: " . join(", ", DBI->available_drivers()), "\n";

my $dbh = DBI->connect("dbi:SQLite:dbname=seinfeld.db", "", "", { RaiseError => 1 });

my $sth = $dbh->prepare("SELECT * FROM foods where name like :1");

$sth->execute('C%');

while($row = $sth->fetchrow_hashref) {
    print @$row{'name'} . "\n";
}

$sth->finish;
$dbh->disconnect;
